### ISSUE TITLE
Improved checking of input suggested

### ISSUE THREAD

**Issue by claczny** [CSB5/OPERA-MS#22]
Hi,

first of all, thank you for your efforts!

I wanted to quickly share an experience I just made with commit cea87dfc79befd962badcab8938af015e8a3797e . I would consider it a bug, but leave the final classification up to you :)

I have my long reads in gzipped format. The packaged version of BLASR seems, however, to not support this. Hence, I decided to use process substitution: `--long-read <(zcat reads.fastq.gz`). OPERA-MS finished without throwing an error and, hence, I thought, all is well :)

I later checked the statistics of the resulting OPERA-MS assembly as well as of the short-read-based assembly that was provided via the `--contig-file` argument and they were very similar (not identical though).
Moreover, I checked the `contig_info.txt` and only there realized that the long read coverage was 0.00 throughout.
That was unexpected and I then tried to specify the long reads as a dedicated file (rather than via process substitution, and, low-and-behold, OPERA-MS failed with the following in ` /mnt/lscratch/users/claczny/ont/testing/results/assembly/operams/lr-sr/intermediate_files/read_mapping/OPERA-long-read.err`

```
/scratch/users/claczny/ont/apps/software/OPERA-MS//utils/blasr  -nproc 28 -m 1 -minMatch 5 -bestn 10 -noSplitSubreads -advanceExactMatches 1 -nCandidates 1 -
maxAnchorsPerPosition 1 -sdpTupleSize 7 /mnt/lscratch/users/claczny/ont/testing/results/assembly/operams/tmp_lr.fna /mnt/lscratch/users
/claczny/ont/testing/results/assembly/megahit/sr/final.contigs.fna | cut -d ' ' -f1-12 | sed 's/ /\t/g' > opera.map 2> blasr.err
[INFO] 2019-08-08T09:37:06 [blasr] started.
sort -k1,1 -k10,10g  opera.map > opera.map.sort
Argument "be" isn't numeric in numeric lt (<) at /scratch/users/claczny/ont/apps/software/OPERA-MS//OPERA-LG/bin//OPERA-long-read.pl line 699, <MAP> line 1.
Argument "one" isn't numeric in subtraction (-) at /scratch/users/claczny/ont/apps/software/OPERA-MS//OPERA-LG/bin//OPERA-long-read.pl line 614, <FILE> line 1.
Argument "of" isn't numeric in subtraction (-) at /scratch/users/claczny/ont/apps/software/OPERA-MS//OPERA-LG/bin//OPERA-long-read.pl line 614, <FILE> line 1.
Argument "be" isn't numeric in division (/) at /scratch/users/claczny/ont/apps/software/OPERA-MS//OPERA-LG/bin//OPERA-long-read.pl line 625.
Illegal division by zero at /scratch/users/claczny/ont/apps/software/OPERA-MS//OPERA-LG/bin//OPERA-long-read.pl line 625.
```

I have now decided to use `--long-read-mapper minimap2` and the job is still running (currently at the `cov_estimate.pl` step).

To summarize: I would suggest to improve the input check routines to avoid these kinds of unexpected events in the future ;) 

Clearly, the user can find out that something strange happened, but it could be made more bold for novice users. 

Accordingly, my suggestion would be to error-out in such a case and, for users who still want to continue, have a respective option that mus be *explicitly* set. After all, maybe some users/scenarios would be fine with this, i.e., consider this a feature rather than a bug ;)

Best,

Cedric

**Comment by dbertran78** [comment]
Hi Cedric, thank you for your feedback. 
As mentioned in the readme, OPERA-MS does not support compressed long-read file, and in the current version their is no file check. Please, re-run your experiment after decompressing the file. 
To avoid such a problem for other users, we will had a file check in our next commit.

**Comment by claczny** [comment]
Thanks for the quick reply!

I was initially positively surprised by the speed, i.e., low execution time, but this is explained by this unexpected behavior now :)
The current run (uncompressed long reads & minimap2) is still running, matching closer what I had expected.

Looking forward to the next release.
